tests/Integration/Momento.Sdk.Tests/AuthClientCacheTest.cs

```diff
@@ -553,4 +550,173 @@ public async Task GenerateDisposableCacheKeyPrefixAuthToken_WriteOnly_HappyPath(
+
+        setResponse = await client.SetAsync(cacheName, "pet-cat", "meow");
+        AssertPermissionError<CacheSetResponse, CacheSetResponse.Error>(setResponse);
+        getResponse = await client.GetAsync(cacheName, "pet-cat");
+        AssertPermissionError<CacheGetResponse, CacheGetResponse.Error>(getResponse);
+    }
+
+    [Fact]
+    public async Task GenerateDisposableMultiPermission_ReadOnlyWithSelectorsAllCaches()
+    {
+        // create a second cache
+        var cache2Name = cacheName + "-2";
+        Assert.True(await cacheClient.CreateCacheAsync(cache2Name) is CreateCacheResponse.Success);
+
+        var scope = new DisposableTokenScope(Permissions: new List<DisposableTokenPermission>{
+            new DisposableToken.CacheItemPermission(
+                CacheRole.ReadOnly,
+                CacheSelector.AllCaches,
+                CacheItemSelector.ByKey("cow")
+            ),
+            new DisposableToken.CacheItemPermission(
+                CacheRole.ReadOnly,
+                CacheSelector.AllCaches,
+                CacheItemSelector.ByKeyPrefix("pet")
+            )
+        });
+        var client = await GetClientForTokenScope(scope);
+
+        // sets should fail for both caches
+        var setResponse = await client.SetAsync(cacheName, "cow", "moo");
+        AssertPermissionError<CacheSetResponse, CacheSetResponse.Error>(setResponse);
+        setResponse = await client.SetAsync(cache2Name, "pet-koala", "awwww");
+        AssertPermissionError<CacheSetResponse, CacheSetResponse.Error>(setResponse);
+
+        // gets should succeed for specified key and key prefix but fail for other keys
+        await SetCacheKey(cacheName, "cow", "moo");
+        await SetCacheKey(cacheName, "pet-koala", "awww");
+        await SetCacheKey(cacheName, "dog", "woof");
+        await SetCacheKey(cache2Name, "cow", "moo");
+        await SetCacheKey(cache2Name, "pet-koala", "awww");
+        await SetCacheKey(cache2Name, "dog", "woof");
+
+        var getResponse = await client.GetAsync(cacheName, "cow");
+        Assert.True(getResponse is CacheGetResponse.Hit);
+        getResponse = await client.GetAsync(cacheName, "pet-koala");
+        Assert.True(getResponse is CacheGetResponse.Hit);
+        getResponse = await client.GetAsync(cache2Name, "cow");
+        Assert.True(getResponse is CacheGetResponse.Hit);
+        getResponse = await client.GetAsync(cache2Name, "pet-koala");
+        Assert.True(getResponse is CacheGetResponse.Hit);
+        getResponse = await client.GetAsync(cacheName, "dog");
+        AssertPermissionError<CacheGetResponse, CacheGetResponse.Error>(getResponse);
+        getResponse = await client.GetAsync(cache2Name, "dog");
+        AssertPermissionError<CacheGetResponse, CacheGetResponse.Error>(getResponse);
+
+        // delete the second cache
+        Assert.True(await cacheClient.DeleteCacheAsync(cache2Name) is DeleteCacheResponse.Success);
```

Comment:
LGTM, but these should go in a finally so we don't leak caches if the test fails.

Reply:
Totally, great point, thanks.

Reply:
Fixed in 3a6928b.